recipes/libgit2/all/conanfile.py

```diff
@@ -0,0 +1,153 @@
+
+        if self.options.with_sha1 == "win32":
+            if self.settings.os != "Windows":
+                raise ConanInvalidConfiguration("win32 is only valid on Windows")
+
+    @property
+    def _need_openssl(self):
+        return "openssl" in (self.options.with_https, self.options.with_sha1)
+
+    @property
+    def _need_mbedtls(self):
+        return "mbedtls" in (self.options.with_https, self.options.with_sha1)
+
+    def requirements(self):
+        self.requires("zlib/1.2.11")
+        self.requires("http_parser/2.9.2")
+        if self.options.with_libssh2:
+            self.requires("libssh2/1.8.2")
+        if self._need_openssl:
+            self.requires("openssl/1.1.1d")
+        if self._need_mbedtls:
+            self.requires("mbedtls/2.16.3-gpl")
+
+    def source(self):
+        tools.get(**self.conan_data["sources"][self.version])
+        extracted_dir = "{}-{}".format(self.name, self.version)
+        os.rename(extracted_dir, self._source_subfolder)
+
+    _cmake_https = {
+        "openssl": "OpenSSL",
+        "winhttp": "WinHTTP",
+        "security": "SecureTransport",
+        "mbedtls": "mbedTLS",
+        False: "OFF",
+    }
+
+    _cmake_sha1 = {
+        "collisiondetection": "CollisionDetection",
+        "commoncrypto": "CommonCrypto",
+        "openssl": "OpenSSL",
+        "mbedtls": "mbedTLS",
+        "generic": "Generic",
+        "win32": "Win32",
+    }
+
+    def _configure_cmake(self):
+        cmake = CMake(self)
+        cmake.definitions["THREADSAFE"] = self.options.threadsafe
+        cmake.definitions["USE_SSH"] = self.options.with_libssh2
+
+        if tools.is_apple_os(self.settings.os):
+            cmake.definitions["USE_ICONV"] = self.options.with_iconv
+        else:
+            cmake.definitions["USE_ICONV"] = True
+
+        cmake.definitions["USE_HTTPS"] = self._cmake_https[str(self.options.with_https)]
+        cmake.definitions["SHA1_BACKEND"] = self._cmake_sha1[str(self.options.with_sha1)]
+
+        cmake.definitions["BUILD_CLAR"] = False
+        cmake.definitions["BUILD_EXAMPLES"] = False
+
+        if self.settings.compiler == "Visual Studio":
+            cmake.definitions["STATIC_CRT"] = "MT" in str(self.settings.compiler.runtime)
+
+        cmake.configure()
+        
+        return cmake
+
+    def _patch_sources(self):
+        tools.replace_in_file(os.path.join(self._source_subfolder, "src", "CMakeLists.txt"),
+                              "FIND_PKGLIBRARIES(LIBSSH2 libssh2)",
+                              "FIND_PACKAGE(libssh2 REQUIRED)\n"
+                              "\tSET(LIBSSH2_FOUND ON)\n"
+                              "\tSET(LIBSSH2_INCLUDE_DIRS ${libssh2_INCLUDE_DIRS})\n"
+                              "\tSET(LIBSSH2_LIBRARIES ${libssh2_LIBRARIES})\n"
+                              "\tSET(LIBSSH2_LIBRARY_DIRS ${libssh2_LIB_DIRS})")
+        tools.save("FindOpenSSL.cmake",
```

Comment:
I think you can use `cmake_find_package` instead. The current generator will provide FindOpenSSL.cmake too

Reply:
hmm, I see that you are already using `cmake_find_package`. Is it not working?

Reply:
The openssl `cmake_find_package` does not define the same variables as the [cmake reference openssl](https://cmake.org/cmake/help/latest/module/FindOpenSSL.html) which are uppercase.

Maybe openssl should use `self.cpp_info.build_modules` to define these variables.

Reply:
Interesting, because I tested on Linux and it's working

Reply:
These lines are (at least) needed on Macos as the build error for that platform is shown.

Reply:
Okay, I didn't check on OSX. Thanks for answering